pkgs/tools/filesystems/sasquatch/default.nix

```
@@ -1,34 +1,45 @@
-{ fetchFromGitHub
+{ lib
+, stdenv
+, fetchFromGitHub
 , fetchurl
-, lz4 ? null
-, lz4Support ? false
-, lzo
-, lib, stdenv
 , xz
+, lzo
 , zlib
+, zstd
+, lz4 ? null
+, lz4Support ? false
 }:
 
 assert lz4Support -> (lz4 != null);
 
 let
-  patch = fetchFromGitHub {
-    owner = "devttys0";
-    repo = "sasquatch";
-    rev = "3e0cc40fc6dbe32bd3a5e6c553b3320d5d91ceed";
-    sha256 = "19lhndjv7v9w6nmszry63zh5rqii9v7wvsbpc2n6q606hyz955g2";
-  } + "/patches/patch0.txt";
+  patch = fetchFromGitHub
+    {
+      # NOTE: This uses my personal fork for now, until
+      # https://github.com/devttys0/sasquatch/pull/40 is merged.
+      # I, cole-h, will keep this fork available until that happens.
+      owner = "cole-h";
+      repo = "sasquatch";
+      rev = "6edc54705454c6410469a9cb5bc58e412779731a";
+      sha256 = "x+PuPYGD4Pd0fcJtlLWByGy/nggsmZkxwSXxJfPvUgo=";
+    } + "/patches/patch0.txt";
```

```suggestion
    # NOTE: This uses my personal fork for now, until
    # https://github.com/devttys0/sasquatch/pull/40 is merged.
    # I, cole-h, will keep this fork available until that happens.
    owner = "cole-h";
    repo = "sasquatch";
    rev = "6edc54705454c6410469a9cb5bc58e412779731a";
    sha256 = "x+PuPYGD4Pd0fcJtlLWByGy/nggsmZkxwSXxJfPvUgo=";
  } + "/patches/patch0.txt";
```

Why are you not using fetchPatch?

---

As per `fetchPatch` documentation:

> Other fetcher functions allow you to add source code directly from a VCS such as subversion or git. These are mostly straightforward names based on the name of the command used with the VCS system.

i.e. can you explicitely have `owner`, `repo`, `rev` when using `fetchPatch`?
It's clearer in this form IMHO, and more convenient when updating...

---

fetchpatch is just a wrapper around fetchurl, so it would be:

```
fetchurl {
  url = "https://raw.githubusercontent.com/cole-h/sasquatch/6edc54705454c6410469a9cb5bc58e412779731a/patches/patch0.txt";
  sha256 = "";
}
```

I like `fetchFromGitHub` better as well.